### PR TITLE
Add tests and fixes for external commit proposal rules

### PR DESCRIPTION
**Pull request by mulmarta**
This makes sure that external joiners can commit custom proposals with custom rules.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 and MIT license.

### CODE REVIEW

File: mls-rs/src/group/external_commit.rs [hunk 1]
```diff
@@ -171,33 +209,45 @@ impl<C: ClientConfig> ExternalCommitBuilder<C> {
...
+
+        #[cfg(all(feature = "custom_proposal", feature = "by_ref_proposal"))]
+        for message in self.received_custom_proposals {
+            let MlsMessagePayload::Plain(plaintext) = message.payload else {
+                return Err(MlsError::UnexpectedMessageType);
+            };
+
+            let auth_content = AuthenticatedContent::from(plaintext.clone());
+
+            verify_plaintext_authentication(&cipher_suite, plaintext, None, None, &group.state)
```
Comment (tomleavy) [inline mls-rs/src/group/external_commit.rs:238]: This wouldn't happen elsewhere in the code?

File: mls-rs/src/group/proposal_filter/filtering_common.rs [hunk 2]
```diff
@@ -482,17 +482,20 @@ fn ensure_exactly_one_external_init(proposals: &ProposalBundle) -> Result<(), Ml
...
-            ProposalType::EXTERNAL_INIT,
-            ProposalType::REMOVE,
-            ProposalType::PSK,
-        ]
-        .contains(ty)
-    });
+    let supported_default_types = [
+        ProposalType::EXTERNAL_INIT,
+        ProposalType::REMOVE,
+        ProposalType::PSK,
```
Comment (tomleavy) [inline mls-rs/src/group/proposal_filter/filtering_common.rs:493]: I was trying to figure out a different time where in the RFC this is defined?

**Review by tomleavy — DISMISSED** [review]

File: mls-rs/src/group/external_commit.rs (hunk 1 above)
Comment (mulmarta) [reply mls-rs/src/group/external_commit.rs:238]: No, this code basically replaces `verify_plaintext_authentication` from the one done by `Group` to the one done by `ExternalGroup` (skip MAC).

So `Group` calls (1) `get_event_from_incoming_message` and then (2) `process_event_or_content` (inside `process_incoming_message_with_time`). And (1) for proposals consists mostly of `verify_plaintext_authentication`. So here we replace (1) with weaker verification and leave (2) intact.

File: mls-rs/src/group/proposal_filter/filtering_common.rs (hunk 2 above)
Comment (mulmarta) [reply mls-rs/src/group/proposal_filter/filtering_common.rs:493]: Doesn't seem to be, which is less than ideal :(

**Review by stefunctional — APPROVED** [review]

File: mls-rs/src/group/external_commit.rs [hunk 3]
```diff
@@ -102,6 +116,31 @@ impl<C: ClientConfig> ExternalCommitBuilder<C> {
...
+    /// Insert a [`CustomProposal`] received from a current group member into the current
+    /// commit that is being built.
+    ///
+    /// # Warning
+    ///
+    /// The authenticity of the proposal is NOT fully verified. It is only verified the
+    /// same way as by [`ExternalGroup`](`crate::external_client::ExternalGroup`).
+    /// The proposal MUST be an MlsPlaintext, else the [`Self::build`] function will fail.
+    pub fn with_received_custom_proposal(mut self, proposal: MlsMessage) -> Self {
+        //use super::message_verifier::verify_plaintext_authentication;
```
Comment (mulmarta) [inline mls-rs/src/group/external_commit.rs:138]:
```suggestion
```

**Review by stefunctional — APPROVED** [review]

**Review by tomleavy — APPROVED** [review]

**Pull request merged**